claim_batch/services.py

```diff
@@ -431,6 +432,109 @@ def _get_capitation_region_and_district(location_id):
 def do_process_batch(audit_user_id, location_id, period, year):
     processed_ids = set()  # As we update claims, we add the claims not in relative pricing and then update the status
     logger.debug("do_process_batch location %s for %s/%s", location_id, period, year)
+    batch_run = new BatchRun
+
+    # 0 prepare the batch run :  does it really make sense per location ? (Ideally per pool but the notion doesn't exist yet)
+    # 0.1 get all product concerned, all product that have are configured for the location
+    _, days_in_month = calendar.monthrange(year, period)
+    end_date = datetime.date(year, period, days_in_month)
+    period_quarter = period - 2 if period|divisibleby:4 else 0
+    period_sem = period - 5 if period|divisibleby:6 else 0
+
+    products = Product.objects\
+            .filter(validity_from__lte = start_date)\
+            .filter(validity_from__gte = end_date)\
+            .filter(validity_to__isnull = True)\
+            .filter(date_from__lte = end_date)\
+            .filter(Q(date_to__gte = start_date) | Q(date_to__isnull = True))\
+            .filter(location_id = location_id)
+    # 1 per product (Ideally per pool but the notion doesn't exist yet)
+    if products:
+        for product in products:
+
+            if product.period_rel_prices = 'Y' or product.period_rel_prices_ip = 'Y' or product.period_rel_prices_op = 'Y':
+                start_date = datetime.date(year, 1, 1)
+            else if product.period_rel_prices = 'S' or product.period_rel_prices_ip = 'S' or product.period_rel_prices_op = 'S':    
+                start_date = datetime.date(year, period_sem, 1)
+            else if product.period_rel_prices = 'Q' or product.period_rel_prices_ip = 'Q' or product.period_rel_prices_op = 'Q':    
+                start_date = datetime.date(year, period_quarter, 1)
+            else:
+                start_date = datetime.date(year, period, 1)
+
+            work_data.product = product
```

Comment:
@delcroip What does mean 'work_data'? i see this is a object that cointains some fields like product, payment plans etc? 
Should be somewhere a definition of representation of this object?

Reply:
I don't think it is required, python should work with this kind of object without class (some fields could be defined but other will be created on the fly by calcules)